android/app/src/main/java/com/mattermost/helpers/DatabaseHelper.kt

```diff
@@ -174,7 +174,8 @@ class DatabaseHelper {
 
                         if (lastId == key) {
                             earliest = post.get("create_at") as Double
-                        } else if (firstId == key) {
+                        }
+                        if (firstId == key) {
```

Comment:
this is the fix for the push notification on Android. Basically if we got only one post from the request, the latest timestamp was not being set and the post not being displayed in the post list.